final_project/snapvisite/models.py

```diff
@@ -60,10 +66,31 @@ def __str__(self):
         return f'({self.company} {self.date})'
 
 
+def get_time_choices():
+    """
+    generates hours from 6am to 11pm, every 30min.
+    returns a list of tuples to choices ( human readable name, actual value)
+    """
+    time_list = []
+    hours = 6
+    minutes = [0, 30]
+    for i in range(1, 37):
+        if i % 2 != 0:
+            time_list.append((f'{time}', time(hours, minutes[0])))
+        if i % 2 == 0:
+            time_list.append((f'{time}', time(hours, minutes[1])))
+        if i % 2 == 0:
+            hours += 1
```

Comment:
1, 37 to tzw. magic numbers, tylko Ty wiesz skąd się wzięły, pasowałoby to przypisać do zmiennych które opisywałyby co to jest, kod się przede wszystkim czyta, pamiętajcie. Nie jestem też zwolennikiem zmiennej `i` to nie jest index, pasuje wymyślić lepszą nazwę